stacktrace_ext_test.go

```diff
@@ -162,3 +165,64 @@ func setupSymlink(t *testing.T, src, dst string) {
 
 	require.NoError(t, os.Symlink(srcAbs, dst), "Failed to set up symlink")
 }
+
+type dependency struct {
+	ImportPath string `json:"Path"` // import path of the dependency
+	Dir        string `json:"Dir"`  // location on disk
+}
+
+// Downloads all dependencies for the current Go module and reports their
+// module paths and locations on disk.
+func downloadDependencies(t *testing.T) []dependency {
+	cmd := exec.Command("go", "mod", "download", "-json")
+
+	stdout, err := cmd.StdoutPipe()
```

Comment:
is streaming worth it here, suggest simplifying to:
```
stdout, err := cmd.Output()
require.NoError(t, err, ...)

dec := json.NewDecoder(bytes.NewReader(stdout))
```

Reply:
Yeah, streaming is necessary here. The output contains a series of newline-delimited JSON objects.

Per `go help mod download`,

```
By default, download reports errors to standard error but is otherwise silent.
The -json flag causes download to print a sequence of JSON objects
to standard output, describing each downloaded module (or failure),
corresponding to this Go struct:

    type Module struct {
        Path     string // module path
        Version  string // module version
        Error    string // error loading module
        Info     string // absolute path to cached .info file
        GoMod    string // absolute path to cached .mod file
        Zip      string // absolute path to cached .zip file
        Dir      string // absolute path to cached source root directory
        Sum      string // checksum for path, version (as in go.sum)
        GoModSum string // checksum for go.mod (as in go.sum)
    }
```

Reply:
We need to use `json.NewDecoder` since the output has multiple objects, but I don't think we need to use a pipe -- we can get all the output at once and then use `json.NewDecoder`

Reply:
Ah, I see what you mean. Switching.